From 85fc62fe9977387dcb3ba0276470b66d6a705ec3 Mon Sep 17 00:00:00 2001
From: Peter Griffin <peter.griffin@linaro.org>
Date: Fri, 20 May 2022 11:47:52 +0100
Subject: [PATCH 5/9] rk3399_common: add dfu_alt_info for rockpi4b

See notes here:
https://git.codelinaro.org/linaro/dependable-boot/meta-ts/-/blob/master/docs/README-rockchip.md
and here:
https://u-boot.readthedocs.io/en/v2022.04/develop/uefi/uefi.html?highlight=efidebug%20capsule%20disk-update#enabling-uefi-capsule-update-feature

On how to generate and apply them.

Signed-off-by: Peter Griffin <peter.griffin@linaro.org>
Signed-off-by: Maxim Uvarov <maxim.uvarov@linaro.org>
---
 include/configs/rk3399_common.h | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/configs/rk3399_common.h b/include/configs/rk3399_common.h
index 2f9aee5819..d089c3f44e 100644
--- a/include/configs/rk3399_common.h
+++ b/include/configs/rk3399_common.h
@@ -26,6 +26,11 @@
 
 #ifndef CONFIG_SPL_BUILD
 
+# define DFU_ALT_INFO \
+        "dfu_alt_info=" \
+        "mmc 1=idbloader.img part 1 1;" \
+        "u-boot.itb part 1 4\0"
+
 #define ENV_MEM_LAYOUT_SETTINGS \
 	"scriptaddr=0x00500000\0" \
 	"script_offset_f=0xffe000\0" \
@@ -49,6 +54,7 @@
 	"fdtfile=" CONFIG_DEFAULT_FDT_FILE "\0" \
 	"partitions=" PARTS_DEFAULT \
 	ROCKCHIP_DEVICE_SETTINGS \
+	DFU_ALT_INFO \
 	BOOTENV \
 	BOOTENV_SF \
 	"altbootcmd=" \
-- 
2.30.2

